strawberryfields/engine.py

```diff
@@ -632,6 +637,8 @@ def run_async(
         kwargs.update(self._backend_options)
 
         device = self.device_spec
+        if program.type == "tdm":
+            device.fill_template(program, device.target)
```

Comment:
@thisac seems like an adjustment is needed here, correct? `target` is derived from the `device` later
```suggestion
            device.fill_template(program)
```

Reply:
Thanks @antalszava. :1st_place_medal: